pkg/apis/core/types.go

```diff
@@ -3017,6 +3030,13 @@ type PodOS struct {
 	Name OSName
 }
 
+// PodSchedulingGate is associated to a Pod to guard its scheduling.
+type PodSchedulingGate struct {
```

Comment:
do we need a struct?

Reply:
Yeah, I asked for it for future proofing.